nixos/modules/services/web-apps/docuseal.nix

```diff
@@ -0,0 +1,174 @@
+    DATABASE_URL = cfg.database_url;
+  } // cfg.extraConfig;
+in
+{
+  options.services.docuseal = {
+    enable = lib.mkEnableOption (lib.mdDoc "DocuSeal, open source document signing");
+
+    database_url = lib.mkOption {
+      type = with lib.types; nullOr str;
+      description = lib.mdDoc ''
+        defaults (`null`) to sqlite. Supports `"postgres:"`/`"mysql:"`
+      '';
+      default = null;
+    };
+
+    secretKeyBaseFile = lib.mkOption {
+      description = lib.mdDoc ''
+        Path to file containing the secret key base.
+        A new secret key base can be generated by running:
+
+        `nix build -f '<nixpkgs>' docuseal; cd result; bin/rake secret`
+
+        If this file does not exist, it will be created with a new secret key base.
+      '';
+      default = "/var/lib/docuseal/secrets/secret-key-base";
+      type = lib.types.path;
+    };
+
+    host = lib.mkOption {
+      description = lib.mdDoc "DocuSeal host.";
+      type = lib.types.str;
+      default = "127.0.0.1";
+    };
+
+    port = lib.mkOption {
+      description = lib.mdDoc "DocuSeal port.";
+      type = lib.types.port;
+      default = 3000;
+    };
+
+    extraConfig = lib.mkOption {
+      type = lib.types.attrs;
+      default = { };
+      description = lib.mdDoc ''
+        Extra environment variables to pass to DocuSeal services.
+      '';
+    };
+
+    extraEnvFiles = lib.mkOption {
+      type = with lib.types; listOf path;
+      default = [ ];
+      description = lib.mdDoc ''
+        Extra environment files to pass to DocuSeal services. Useful for passing down environmental secrets.
+      '';
+      example = [ "/etc/docuseal/s3config.env" ];
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.docuseal;
+      defaultText = lib.literalExpression "pkgs.docuseal";
+      description = lib.mdDoc "DocuSeal package to use.";
+    };
+
+    user = lib.mkOption {
+      description = lib.mdDoc ''
+        User under which docuseal runs. If it is set to "docuseal",
+        that user will be created, otherwise it should be set to the
+        name of a user created elsewhere.
+      '';
+      type = lib.types.str;
+      default = "docuseal";
+    };
+
+    group = lib.mkOption {
+      description = lib.mdDoc ''
+        Group under which docuseal runs.
+      '';
+      type = lib.types.str;
+      default = "docuseal";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.docuseal = {
+      description = "DocuSeal server";
+      wantedBy = [ "multi-user.target" ];
+      environment = env;
+      serviceConfig = {
+        Type = "simple";
+        ExecStartPre = pkgs.writeShellScript "docuseal-pre-script" ''
+          cat > /var/lib/docuseal/docuseal.env <<EOF
+          SECRET_KEY_BASE="$(cat ${cfg.secretKeyBaseFile})"
+          EOF
+        '';
+        ExecStart = "${cfg.package}/bin/rails server --pid=/var/lib/docuseal/docuseal.pids";
```

Comment:
I'm not a systemd expert, but I wonder if that `--pid` flag is necessary?

Reply:
the `--pid` flag is for ruby. If this is not present, then ruby want to place it next to its application ( into `/nix/store/`)